actionpack/lib/action_dispatch/journey/visitors.rb

```diff
@@ -185,7 +186,7 @@ def terminal(node, seed)
           end
 
           def visit_GROUP(node, seed)
-            visit(node.left, seed << "(".freeze) << ")".freeze
+            visit(node.left, seed.dup << "(".freeze) << ")".freeze
```

Comment:
Good idea! I didn't want to do that at first, but now when we have `# frozen_string_literal: true` in the updated files it makes sense to do it.

I updated the PR.